sonar-batch/src/main/java/org/sonar/batch/cache/WSLoader.java

```diff
@@ -156,33 +143,37 @@ private void updateCache(String id, byte[] value) {
     }
   }
 
+  interface DataLoader<T> {
+    T load(String id) throws IOException;
+  }
+
   @Nonnull
-  private WSLoaderResult<byte[]> loadFromCache(String id) throws NotAvailableException {
+  private <T> WSLoaderResult<T> loadFromCache(String id, DataLoader<T> loader) throws NotAvailableException {
+    T result = null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "result". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1854)